### ISSUE TITLE
[2.5.*] tns run always add latest platform 

### ISSUE THREAD

**Issue by tsonevn** [NativeScript/nativescript-cli#2782]
_From @3rror404 on May 4, 2017 15:52_

I'm still using N 2.5.4 as I don't have time to upgrade to version 3 yet.

After I removed the iOS platform and re-added I noticed that the platform had been upgraded to 3.0.0.  After removing the platform again and reinstalling the 2.5.0 version with `tns platform add ios@2.5.0` I can no longer run my project.

```
Project successfully prepared (ios)
Searching for devices...
Building project...
Expected "// !$*UTF8*$!" but "<" found.
# run ios
```

I've tried numerous things including removing iOS and installing the 3.0.0 version. Removing the hooks, node_modules and platforms folders all with the same result.

Any ideas?

```
┌──────────────────┬─────────────────┬────────────────┬──────────────────┐
│ Component        │ Current version │ Latest version │ Information      │
│ nativescript     │ 2.5.4           │ 3.0.0          │ Update available │
│ tns-core-modules │ 2.5.2           │ 3.0.0          │ Update available │
│ tns-android      │ 3.0.0           │ 3.0.0          │ Up to date       │
│ tns-ios          │ 2.5.0           │ 3.0.0          │ Update available │
└──────────────────┴─────────────────┴────────────────┴──────────────────┘
```

_Copied from original issue: NativeScript/NativeScript#4115_

**Comment by tsonevn** [comment]
Hi @3rror404 ,
Thank you for reporting this issue.

I tested it on my side and I have to confirm that this is a real issue, while using an old CLI.
If we have an old project, using runtime 2.5.0 and build with `tns run <platform_name>` tns-ios` and `tns -android` will be automatically updated to version 3.0.

Workaround:
1. `tns platform remove <platform_name>`
2. `tns platform add <platform_name>`
3. `tns prepare <platform_name>`
4. `tns run <platform_name>`

**Comment by Cae2** [comment]
I think that this workaround isnt working with me.. I'm using tns 2.5.4 and it is still installing tns-android 3

![3 0](https://cloud.githubusercontent.com/assets/24416663/25747939/af86a648-317f-11e7-8666-473b8eca1dd6.jpg)

**Comment by 3rror404** [comment]
Same here.

**Comment by tsonevn** [comment]
Hi @Cae2, @3rror404
Excuse me for the mistake. I forgot to specify the runtime version while adding the platform. For example: `tns platform add <platform_name>@2.5.0`. I also update my previous answer.
Now the workaround should work for you.

**Comment by 3rror404** [comment]
I did actual try that thinking that you might have made a mistake and I still received the same error:

```
Expected "// !$*UTF8*$!" but "<" found.
```

**Comment by m-abs** [comment]
The workaround doesn't work when I try publish a bundled version.
**ns-bundle** removes the platform and adds it against, so I'm still have tns-android@3.0.0.

**Comment by dtopuzov** [comment]
@m-abs, @3rror404, @Cae2 Fix is available in {N} 2.5.5
```
npm un -g nativescript
npm cache clean
npm i -g nativescript@2.5.5
```
